src/oc_erchef/apps/oc_chef_authz/src/oc_chef_authz_scoped_name.erl

```diff
@@ -0,0 +1,495 @@
+
+parse_scoped_names(Names, Context) ->
+    parse_scoped_names(Names, true, Context).
+
+parse_unscoped_names(Names, Context) ->
+    parse_scoped_names(Names, false, Context).
+
+-spec parse_scoped_names([binary()], boolean(), #context{}) -> [#sname{} | {atom(), binary()}].
+parse_scoped_names(Names, ScopedOk, Context) ->
+    Pattern = make_regex(),
+    [ maybe_parse_scoped_name(Name, Pattern, ScopedOk, Context) || Name <- Names ].
+
+-spec maybe_parse_scoped_name(binary(), re:mp(), boolean(), #context{}) ->
+                                     #sname{} | {ill_formed_name|inappropriate_scoped_name, binary()}.
+maybe_parse_scoped_name(Name, Pattern, ScopedOk, Context) ->
+    process_match(re:run(Name, Pattern, [{capture, all, binary}]), Name, Context, ScopedOk).
+
+%% Process the various formats, and expand them
+%%
+%% If no match, then we have bad name
+process_match(nomatch, Name, _Context, _ScopedOk) ->
+    {ill_formed_name, Name};
+%% If we only onl match the name, then it is an unscoped name, and we use the org context
+process_match({match, [Name, Name]}, Name, Context, _ScopedOk) ->
+    set_org_from_context(#sname{base = Name, full = Name}, Context);
+%% Anything not a simple, unqualified name should be rejected
+process_match({match, _}, Name, _, false) ->
+    {inappropriate_scoped_name, Name};
+%% If scope is omitted, assume global
+process_match({match, [_, <<>>, <<>>, BaseName]}, Name, _, true) ->
+    #sname{org = global_org, base = BaseName, full = Name};
+%% Fully qualified name
+process_match({match, [_, <<>>, OrgName, BaseName]}, Name, _, true) ->
+    #sname{org = OrgName, base = BaseName, full = Name}.
+
+set_org_from_context(#sname{} = ScopedName, #context{org_id = OrgId}) when OrgId =/= undefined ->
+    ScopedName#sname{org_id = OrgId}.
+
+%%
+%% Takes a list of {K, V} pairs with multiple instances of K, and regroups them.
+%%
+-spec group_by_key([{any(), any()}]) -> [{any(), [any()]}].
+group_by_key(L) ->
+    Map = lists:foldl(fun({K, V}, Map) ->
+                              VL = maps:get(K, Map, []),
+                              maps:put(K, [V | VL], Map)
+                      end,
+                      #{}, L),
+    maps:to_list(Map).
+
+%%
+%%
+make_sql_callback() ->
+    fun chef_sql:select_rows/1.
+
+%%
+%% Helper function for working with unscoped names
+%%
+-spec make_scoped_names(binary(), [binary()]) -> [{binary(), binary()}].
+make_scoped_names(OrgName, Names) ->
+    [{OrgName, Name} || Name <- Names].
+
+%%
+%% Expansion of authz ids into scoped names
+%% Takes {OrgName, Name} pairs in ScopedNames and returns
+%% list of names with scoping metacharacter inserted
+-spec render_names_in_context(binary(),[{binary(), [binary()]}], #context{}) -> [binary()].
+render_names_in_context(OrgId, ScopedNames, Context) ->
+    GroupedScopedNames = group_by_key(ScopedNames),
+    {Expanded, _Cache} = lists:foldl(fun(E, A) -> render_names_in_context_f(OrgId, E, A) end,
```

Comment:
We do not use the `Context` in this fun. This is the only place we consume `render_names_in_context_f` and we drop it in `_Cache`. The fun itself is not using it so maybe we should remove it?